daskhub/README.md

```diff
@@ -0,0 +1,177 @@
+
+- A standard Dask Gateway deployment using the Dask Gateway helm chart
+  configured to use JupyterHub for authentication.
+- A standard JupyterHub deployment using the JupyterHub helm chart.
+
+## Prepare Configuration File
+
+In this step, we'll prepare a YAML configuration file with the fields
+required by the DaskHub helm chart. It will contain some secret
+keys, which should not be checked into version control in plaintext.
+
+We need two random hex strings that will be used as keys, one for
+JupyterHub and one for Dask Gateway.
+
+
+Run the following command, and copy the output. This is our `token-1`.
+
+```console
+openssl rand -hex 32
+```
+
+Run command again and copy the output. This is our `token-2`.
+
+```console
+openssl rand -hex 32
+```
+
+Now substitute those two values for `<token-1>` and `<token-2>` below.
+Note that `<token-2>` is used twice, once for `jupyterhub.hub.services.dask-gateway.apiToken`, and a second time for `dask-gateway.gateway.auth.jupyterhub.apiToken`.
+
+
+```yaml
+# secrets.yaml
+jupyterhub:
+  proxy:
+    secretToken: "<token-1>"
+  hub:
+    services:
+      dask-gateway:
+        apiToken: "<token-2>"
+
+dask-gateway:
+  gateway:
+    auth:
+      jupyterhub:
+        apiToken: "<token-2>"
+```
+
+## Install Dask Hub
+
+This example installs into the namespace `dhub`. Make sure you're
+in the same directory as the `secrets.yaml` file.
+
+```console
+$ helm upgrade --wait --install --render-subchart-notes \
+    dhub dask/daskhub \
+    --namespace=dhub \
+    --version=0.0.1 \
+    --values=secrets.yaml
+```
+
+The output explains how to find the IPs for your JupyterHub and Dask Gateway.
+
+```console
+$ kubectl -n dhub get service
+NAME                                    TYPE           CLUSTER-IP      EXTERNAL-IP     PORT(S)                      AGE
+api-us-central1b-dhub-dask-gateway      ClusterIP      10.43.253.126   <none>          8000/TCP                     25m
+dask-e10f1f0179784a60b0e4c5ba09c1ed44   ClusterIP      None            <none>          8786/TCP,8787/TCP,8788/TCP   100s
+hub                                     ClusterIP      10.43.250.8     <none>          8081/TCP                     25m
+proxy-api                               ClusterIP      10.43.248.47    <none>          8001/TCP                     25m
+proxy-public                            LoadBalancer   10.43.244.163   35.224.253.72   443:31356/TCP,80:32352/TCP   25m
+traefik-us-central1b-dhub-dask-gateway  LoadBalancer   10.43.244.244   35.223.8.79     80:31076/TCP                 25m
+```
+
+JupyterHub is available at the `proxy-public` external ip (35.224.253.72 in this example).
+Dask Gateawy is available at `traefik-us-central1b-dhub-dask-gateway` (35.223.8.79).
+
+## Creating a Dask Cluster
+
+To create a Dask cluster, connect to the Dask Gateway
+
+```python
+>>> from dask_gateway import Gateway
+>>> gateway = dask_gateway.Gateway(
+...     address="http://35.223.8.79",  # traefik-us-central1b-dhub-dask-gateway
```

Comment:
Are we not configuring this automatically?

Reply:
Unfortunately not :( In the "Configuring JupyterHub" section I give some details for how the administrator can set it for JupyterHub users.

If we can figure out how to set helm templated environment variables for the user then I think we can make this work.

Reply:
Ah right this is the template problem from earlier.